lib/addons/fluent-bit/index.ts

```diff
@@ -0,0 +1,117 @@
+import * as eks from '@aws-cdk/aws-eks';
+import * as es from '@aws-cdk/aws-elasticsearch';
+import { Effect, PolicyStatement } from '@aws-cdk/aws-iam';
+
+import { Constants } from "..";
+import { ClusterInfo } from "../../spi/types";
+import { Team } from "../../spi/team-contracts";
+import { ClusterAddOn } from "../../spi/addon-contracts"
+import { createNamespace } from "../../utils/namespace";
+
+/**
+ * Configuration options for the FluentBit add-on.
+ */
+export interface FluentBitAddOnProps {
+    /**
+     * The namespace for the FluentBit add-on.
+     * @default `logging`
+     */
+    readonly namespace: string;
+
+    /**
+     * Props for Elasticsearch support.
+     */
+    readonly elasticsearch?: es.DomainAttributes
+}
+
+/**
+ * Default props for the add-on.
+ */
+const defaultProps: FluentBitAddOnProps = {
+    namespace: 'logging',
+}
+
+/**
+ * FluentBitAddOn deploys FluentBit into an EKS cluster using the `aws-for-fluent-bit` Helm chart.
+ * https://github.com/aws/eks-charts/tree/master/stable/aws-for-fluent-bit
+ */
+export class FluentBitAddOn implements ClusterAddOn {
+
+    readonly props: FluentBitAddOnProps
+
+    constructor(props?: FluentBitAddOnProps) {
+        this.props = { ...defaultProps, ...props }
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+        // Create the FluentBit namespace.
+        const cluster = clusterInfo.cluster;
+        const namespace = this.props.namespace
+        createNamespace(cluster, namespace)
+
+        // Create the FluentBut service account.
+        const serviceAccountName = 'fluent-bit-service-account'
+        const sa = cluster.addServiceAccount(serviceAccountName, {
```

Comment:
Need a dependency between sa and namespace, otherwise intermittent failures on race conditions.